### ISSUE TITLE
General: Public analyses table format

### ISSUE THREAD

**Issue by danicahelb** [VEuPathDB/web-eda#448]
On the EDA site, the Public Analyses link should take you to a stand-alone “Public analyses” table

This is in place of having the Public analyses table live within a tab on “My search strategies” (this is where the current Public strategies table lives in the WDK site)

Structure of the Public analyses table will look similar to the My analysis table currently on the EDA site, and will have the following columns in the following order:  
1. **Study:** Study label
2. **Analysis:** user-provided name given to the analysis
3. **Description:** user-provided description of the analysis
4. **Author:** who created the analysis
5. **Organization:** affiliated institution for the author of the analysis
6. **Created:** when the analysis was created
7. **Modified:** when the analysis was modified

**Comment by danicahelb** [comment]
This is dependent on https://github.com/VEuPathDB/web-eda/issues/441

**Comment by danicahelb** [comment]
@jtlong3rd we would like to make the following improvements to the public analyses table, as we discussed in 10/1 UX meeting:

1. Allow the author to be able to edit the analysis name directly within the table (implement pencil icon)
2. Remove links to studies in the “Study” column
3. 1st column = Study, 2nd column = Analysis
4. Make all columns searchable (currently, just “Study” and “Analysis” columns are searchable)

This was not discussed in the UX meeting, but I was wondering if it would be possible to add a column of pins so that users could pin public analyses of interest to the top of the table? this would have to be user specific, and would only show up once the user was logged in.

**Comment by jtlong3rd** [comment]
> @jtlong3rd we would like to make the following improvements to the public analyses table, as we discussed in 10/1 UX meeting:
> 
> 1. Allow the author to be able to edit the analysis name directly within the table (implement pencil icon)
> 2. Remove links to studies in the “Study” column
> 3. 1st column = Study, 2nd column = Analysis
> 4. Make all columns searchable (currently, just “Study” and “Analysis” columns are searchable)
> 
> This was not discussed in the UX meeting, but I was wondering if it would be possible to add a column of pins so that users could pin public analyses of interest to the top of the table? this would have to be user specific, and would only show up once the user was logged in.

@danicahelb Yes, it would definitely be possible to add user-specific "pinned public analyses" to the UI - this would be a pure client change (no user service updates required). Let's run this idea by a wider group.

**Comment by jtlong3rd** [comment]
> @jtlong3rd @danicahelb i am not sure the pinning should be phase 1. but jamie, i suppose this could be a soft user preference managed by the client? it could be a list of analysis IDs?

Yup, makes sense. We could also make use of the [EDA user service preferences](https://veupathdb.github.io/EdaUserService/api.html#resources:/users/{user-id}/preferences:get). (Currently, those  preferences are a JSON string with a single optional `pinnedAnalyses` property, which consists of a list of analysis IDs. The client could be updated to save/read an optional `pinnedPublicAnalyses` field that behaves similarly.)

**Comment by danicahelb** [comment]
@jtlong3rd the table looks great, but i'm not following the conversation around pinning. I'm going to close this ticket and create a new ticket on the pinning, which can get implemented in phase 2: https://github.com/VEuPathDB/EdaClinepiPhase2NewIssues/issues/41

**Comment by jtlong3rd** [comment]
> @jtlong3rd the table looks great, but i'm not following the conversation around pinning. I'm going to close this ticket and create a new ticket on the pinning, which can get implemented in phase 2: [VEuPathDB/EdaClinepiPhase2NewIssues#41](https://github.com/VEuPathDB/EdaClinepiPhase2NewIssues/issues/41)

Sounds like a plan! The gist of the dialogue around pinning is that we shouldn't need to do any new service work; just client work.
